tests/falcon_test.py

```diff
@@ -88,3 +119,32 @@ class HelloResource:
         spec.path(resource=hello_resource)
 
         assert spec._paths["/hi"]["x-extension"] == "global metadata"
+
+    def test_unredundant_basepath_resource(self, app, spec_factory, settings):
+        class HelloResource:
+            def on_get(self, req, resp):
+                """A greeting endpoint.
+                ---
+                description: get a greeting
+                responses:
+                    200:
+                        description: said hi
+                """
+                return "dummy"
+
+        expected = {
+            "description": "get a greeting",
+            "responses": {"200": {"description": "said hi"}},
+        }
+        full_path = "/v1/foo/v1"
+        path_expected = "/foo/v1"
+        hello_resource = HelloResource()
+        app.add_route(full_path, hello_resource)
+        spec = spec_factory(app)
+        base_path = \
+            settings["servers"][0]["variables"]["basePath"].get("default")
+        # in swagger 2 this is simply 'basePath: /v1'
```

Comment:
Could you expand on this comment?

If the behavior is different for different versions, this might be something we should include in the `README.md` versus a comment in a test.

Reply:
I've put comment in the code.

As I read the spec exampes, basePath in swagger 2 usually comes with leading slash, but not in openapi 3, they usually in plain path name. But in case people interchange the two it will not be a problem.

Yes, I think it I'd be better if it explained in the README file.

Reply:
Thanks for adding the comment in the code!!

Do you think it makes sense to add a test for the version 3 variant to ensure functionality does not regress?

I can update the README when I cut a release this weekend.